.infrastructure/helm-chart/todoapp/templates/secret.yml

```diff
@@ -0,0 +1,10 @@
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ .Chart.Name }}-secret
+  namespace: {{ .Values.todoapp.namespace }}
+type: Opaque
+data:
+  {{- range $k, $v := .Values.todoapp.secrets}}
+  {{ $k }}: {{ $v }}
```

Comment:
Ensure that the values in `.Values.todoapp.secrets` are base64 encoded. Kubernetes requires secret data to be encoded in base64, so verify that the values provided in your `values.yaml` file are correctly encoded.